sql/core/src/test/scala/org/apache/spark/sql/ExpressionsSchemaSuite.scala

```diff
@@ -152,23 +152,37 @@ class ExpressionsSchemaSuite extends QueryTest with SharedSparkSession {
 
     val outputSize = outputs.size
     val headerSize = header.size
-    val expectedOutputs: Seq[QueryOutput] = {
+    val (expectedMissingExamples, expectedOutputs) = {
       val expectedGoldenOutput = fileToString(resultFile)
       val lines = expectedGoldenOutput.split("\n")
       val expectedSize = lines.size
 
       assert(expectedSize == outputSize + headerSize,
         s"Expected $expectedSize blocks in result file but got " +
-          s"${outputSize + headerSize}. Try regenerate the result files.")
+          s"${outputSize + headerSize}. Try regenerating the result files.")
 
-      Seq.tabulate(outputSize) { i =>
+      val numberOfQueries = lines(2).split(":")(1).trim.toInt
+      val numberOfMissingExamples = lines(3).split(":")(1).trim.toInt
+      val missingExamples = lines(4).split(":")(1).trim.split(",")
+      val expectedOutputs = Seq.tabulate(outputSize) { i =>
         val segments = lines(i + headerSize).split('|')
         QueryOutput(
           className = segments(1).trim,
           funcName = segments(2).trim,
           sql = segments(3).trim,
           schema = segments(4).trim)
       }
+
+      // Ensure consistency of the result file.
+      assert(numberOfQueries == expectedOutputs.size,
```

Comment:
I think should put  these assert on line 163

Reply:
`numberOfQueries == outputSize`

Reply:
@beliefer This assert place here to verify the consistency of the file contents to avoid inconsistency caused by manual modification, and I think line 189  `assert expectedOutputs.size == outputSize` achieves the same goal as `numberOfQueries == outputSize`. Is this acceptable?

Reply:
You can try to mock a new function with comments and test.

Reply:
With a a new function add scene, the right way is run this case with `SPARK_GENERATE_GOLDEN_FILES = 1` to automatically regenerate the correct `sql-expression-schema.md`.

The `sql-expression-schema.md` header said ` Automatically generated by ExpressionsSchemaSuite`.

Reply:
Therefore, if the file is manually modified instead of automatically generated,  I think the assertion failure caused by incorrect modification should be expected.

Do we need to tell users clearly with `Try regenerating the result files with sys env  SPARK_GENERATE_GOLDEN_FILES = 1`?

Reply:
Address [6e6489b](https://github.com/apache/spark/pull/29608/commits/6e6489bff1291cf5d409b04b7dd21e56daf88e3c) reorder the assertions.